src/db/collections/block.rs

```diff
@@ -313,6 +320,15 @@ impl MongoDb {
                         "block.payload.transaction_id": &output_id.transaction_id,
                         "$expr": { "$gt": [{ "$size": "$block.payload.essence.outputs" }, &(output_id.index as i64)] }
                     } },
+                    doc! { "$lookup": {
+                        "from": MilestoneDocument::COLLECTION,
+                        "pipeline": [
+                            { "$sort": { "milestone_timestamp": -1 } },
+                            { "$limit": 1 }
```

Comment:
out of interest: why not just use the `milestone_index` (protocol internal clock)? Due to the indexing there's no performance difference, right, and it becomes a matter of taste?

Reply:
We use the `milestone_index` in most places, so it might make sense to unify. Ultimately, it's up to you @Alexandcoats.